loki/transformations/inline.py

```diff
@@ -389,6 +393,19 @@ def __init__(self, query, recurse_query=None, inline_elementals_only=False,
         def map_inline_call(self, expr, *args, **kwargs):
             if not self.visit(expr, *args, **kwargs):
                 return
+            if expr.procedure_type is BasicType.DEFERRED or isinstance(expr.routine, StatementFunction):
+                return
+            if self.inline_elementals_only:
+                if not (expr.procedure_type.is_function and expr.procedure_type.is_elemental):
+                    return
+            if functions:
+                if expr.routine not in functions:
+                    return
+            if (expr.procedure_type.is_function and expr.procedure_type.is_elemental):
```

Comment:
And this would simplify to
```suggestion
            if expr.procedure_type.is_elemental:
```